Jenkinsfile

```diff
@@ -271,6 +271,30 @@ def getuid() {
     return cached_uid
 }
 
+// Default priority is 3, lower is better.
+// The parameter for a job is set using the script/Jenkinsfile from the
+// previous build of that job, so the first build of any PR will always
+// run at default because Jenkins sees it as a new job, but subsequent
+// ones will use the value from here.
+// The advantage therefore is not to change the priority of PRs, but to
+// change the master branch itself to run at lower priority, resulting
+// in faster time-to-result for PRs.
+
+String get_priority() {
+    node {
+        checkoutScm()
+        if (cachedCommitPragma(pragma: 'Queue-jump', def_val: false)) {
+	    string p = '2'
+	} else if (env.BRANCH_NAME == 'master') {
+	    string p = '4'
+	} else {
+	    string p = ''
+	}
+	echo "Build priority set to " + p
+	return p
+    }
```

Comment:
@ashleypittman any thoughts?

Reply:
I incorporated the style changes etc, and removed the commit pragma which were the technical comments.

The remaining difference between our opinions that I'm aware of is of the benefit, obviously this PR isn't magic and doesn't change the amount of work that Jenkins has to perform so it's down to a decision of if the trade-off is worthwhile.  I'm for it because time-to-result is important but it does have a cost.

Reply:
Indeed.  This PR is fine in it's current condition and I will be +1ing it for it's technical merit.

I just question whether we are willing to accept a reduced time-to-result at the expense of not running any master landings for what my gut suggests to me might be days.  It could end up being only once or twice a week.